pkg/controller/usercleanup/user_cleanup_controller.go

```diff
@@ -0,0 +1,166 @@
+package usercleanup
+
+import (
+	"context"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	crtCfg "github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+var log = logf.Log.WithName("controller_usercleanup")
+
+type StatusUpdater func(userAcc *toolchainv1alpha1.UserSignup, message string) error
+
+// Add creates a new UserCleanup Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, crtConfig *crtCfg.Config) error {
+	return add(mgr, newReconciler(mgr, crtConfig))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, crtConfig *crtCfg.Config) reconcile.Reconciler {
+	return &ReconcileUserCleanup{
+		client:    mgr.GetClient(),
+		scheme:    mgr.GetScheme(),
+		crtConfig: crtConfig,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("usercleanup-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the secondary resource UserSignup and requeue the owner UserSignup
+	if err := c.Watch(
+		&source.Kind{Type: &toolchainv1alpha1.UserSignup{}},
+		&handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &toolchainv1alpha1.UserSignup{},
+		}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileUserCleanup{}
+
+// ReconcileUserCleanup cleans up old UserSignup resources
+type ReconcileUserCleanup struct {
```

Comment:
Could we name the file, directory and the controller `UserSignupCleanup` to make clear that it cleans just the `UserSignups` and not the entire user accounts.

Reply:
Done